ngraph_bridge/ngraph_builder.cc

```diff
@@ -1065,55 +1063,26 @@ static Status TranslateDepthwiseConv2dNativeOp(
                        ng_strides, ng_dilations, ng_padding_below,
                        ng_padding_above);
 
-  // ng input shape is NCHW
-  auto& input_shape = ng_input.get_shape();
-  // ng filter shape is OIHW
-  auto& filter_shape = ng_filter.get_shape();
-  ng::OutputVector ng_args;
-
-  for (size_t i = 0; i < input_shape[1]; i++) {
-    const std::vector<size_t> lower_bound_vec{0, i, 0, 0};
-    const std::vector<size_t> upper_bound_vec{input_shape[0], i + 1,
-                                              input_shape[2], input_shape[3]};
-    auto lower_bound = ConstructNgNode<opset::Constant>(
-        op->name(), ng::element::i64, ng::Shape{lower_bound_vec.size()},
-        lower_bound_vec);
-    auto upper_bound = ConstructNgNode<opset::Constant>(
-        op->name(), ng::element::i64, ng::Shape{upper_bound_vec.size()},
-        upper_bound_vec);
-    auto ng_sliced_input = ConstructNgNode<opset::StridedSlice>(
-        op->name(), ng_input, lower_bound, upper_bound, std::vector<int64_t>{},
-        std::vector<int64_t>{});
-
-    const std::vector<size_t> f_lower_bound_vec{0, i, 0, 0};
-    const std::vector<size_t> f_upper_bound_vec{
-        filter_shape[0], i + 1, filter_shape[2], filter_shape[3]};
-    auto f_lower_bound = ConstructNgNode<opset::Constant>(
-        op->name(), ng::element::i64, ng::Shape{f_lower_bound_vec.size()},
-        f_lower_bound_vec);
-    auto f_upper_bound = ConstructNgNode<opset::Constant>(
-        op->name(), ng::element::i64, ng::Shape{f_upper_bound_vec.size()},
-        f_upper_bound_vec);
-    auto ng_sliced_filter = ConstructNgNode<opset::StridedSlice>(
-        op->name(), ng_filter, f_lower_bound, f_upper_bound,
-        std::vector<int64_t>{}, std::vector<int64_t>{});
-
-    NGRAPH_VLOG(3) << "depthwise conv 2d.";
-    NGRAPH_VLOG(3) << "sliced shape " << ng::join(ng_sliced_input.get_shape());
-    NGRAPH_VLOG(3) << "filter shape " << ng::join(ng_sliced_filter.get_shape());
-    auto ng_conv = ConstructNgNode<opset::Convolution>(
-        op->name(), ng_sliced_input, ng_sliced_filter, ng_strides,
-        ng_padding_below, ng_padding_above, ng_dilations);
-
-    ng_args.push_back(ng_conv);
-  }
-
-  size_t ng_concatenation_axis = 1;  // channel axis
-  auto ng_concat = ConstructNgNode<opset::Concat>(op->name(), ng_args,
-                                                  ng_concatenation_axis);
+  // H W I M -> H W I 1 M
```

Comment:
Is there an existing test, if not can you please add a test?

Reply:
Yes, there are tests in `test/python/test_depthwiseconv2d.py` for both single and multiple groups that I used to debug the translation. Besides, few of the models enabled in the `models-cpu` pipeline also cover the translation now.